### PR TITLE
Admin volunteer dashboard - proof of concept

### PR DESCRIPTION
**Pull request by Holben888**
Adds the admin dashboard component with dummy user data to display.
Notable additions:
- Slight change to root layout to fill screen height and remove unnecessary wrapper `div`s
- New admin dash layout with sidebar and applicant info panel
- Filter modal, mostly for show

Future changes to make:
- Move each css module from the "styles" folder to the same folder as the component using it
- Bridge over SCSS variables to be used in custom modules
- Clean up layouts to satisfy Prettier and avoid unneeded nesting (didn't have time for this)

### CODE REVIEW

File: frontend/src/components/AdminDash/AppInfo.jsx [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+      weekday_afternoons: true,
+      weekday_evenings: false,
+      weekend_mornings: false,
+      weekend_afternoons: true,
+      weekend_evenings: false
+    }
+  }
+];
+
+export default class AppInfo extends Component {
```
Comment (ahijaouy) [inline frontend/src/components/AdminDash/AppInfo.jsx]: Seems like this component should maybe be a functional presentational component, as opposed to being a class

Comment (ahijaouy) [reply frontend/src/components/AdminDash/AppInfo.jsx]: Also should `userData` be passed in as a prop into `AppInfo`?

File: frontend/src/components/AdminDash/AppInfo.jsx [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+              </p>
+            </Col>
+          </Row>
+          <Row>
+            <Col>
+              <h5 className={styles.label}>
+                <Label for="exampleAddress">Address</Label>
+              </h5>
+              <p className={styles.content}>
+                {userData[0].bio.street_address + " " + userData[0].bio.city + ", " + userData[0].bio.state + " " + userData[0].bio.zip_code}
```
Comment (ahijaouy) [inline frontend/src/components/AdminDash/AppInfo.jsx]: Maybe a Template string would be cleaner here?

File: frontend/src/components/AdminDash/AppInfo.jsx [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+                <Label for="exampleEmail">Email</Label>
+              </h5>
+              <p className={styles.content}>{email}</p>
+            </Col>
+            <Col>
+              <h5 className={styles.label}>
+                <Label for="examplePhoneNumber">Phone Number</Label>
+              </h5>
+              <p className={styles.content}>
+                {userData[0].bio.phone_number}
```
Comment (ahijaouy) [inline frontend/src/components/AdminDash/AppInfo.jsx]: Eventually we wouldn't wanna index into `userData[0]` like this right? We'd either be mapping over an array or just getting an object in `AppInfo`?

File: frontend/src/components/AdminDash/index.js [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+  constructor() {
+    super();
+    this.state = {
+      selectedApplicantIndex: 0,
+      showFilterModal: false
+    };
+    this.onSelectApplicant = this.onSelectApplicant.bind(this);
+    this.onShowFilterModal = this.onShowFilterModal.bind(this);
+  }
+  onSelectApplicant(index) {
```
Comment (ahijaouy) [inline frontend/src/components/AdminDash/index.js]: This approach is gucci, but just for information, if you don't wanna do the explicit bind in the constructor you can do something like this
```js
onSelectApplicant = index => {...}
```

And then it'll automatically be bound and you'll be gucci :) But this is also a good way of doing it!

File: frontend/src/components/AdminDash/AppInfo.jsx (hunk 1 above)
Comment (ahijaouy) [reply frontend/src/components/AdminDash/AppInfo.jsx]: Also, seems like we're calling this same component `AppInfo` and `ApplicantInfo` interchangeably in our code base. Might suggest we pick a name and stick with it. Personally I like `ApplicantInfo` :)

File: frontend/src/styles/scss/custom.scss [hunk 5]
```diff
@@ -1,18 +1,15 @@
-
 // Your variable overrides
 // $body-bg: #000;
-$body-bg:  rgb(194, 239, 255);
+$body-bg : rgb(194, 239, 255);
```
Comment (ahijaouy) [inline frontend/src/styles/scss/custom.scss]: lol typo?

File: frontend/src/components/AdminDash/Filters.jsx [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+import { Button, Form, FormGroup, Modal, ModalHeader, ModalBody, ModalFooter, CustomInput, Label } from 'reactstrap';
+
+class Filters extends React.Component {
+  render() {
+    const { show, toggleCallback } = this.props;
+    return (
+      <Modal isOpen={show}>
+        <ModalHeader toggle={toggleCallback}>Filters</ModalHeader>
+        <ModalBody>
+          <FormGroup>
```
Comment (ahijaouy) [inline frontend/src/components/AdminDash/Filters.jsx]: Do the inputs on this form get submitted anywhere rn? If so may eventually wanna check out the `formik` components I made and see if they might be useful here

File: frontend/src/components/AdminDash/index.js [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+  onShowFilterModal() {
+    this.setState({
+      showFilterModal: !this.state.showFilterModal
+    });
+  }
+  render() {
+    const { selectedApplicantIndex, showFilterModal } = this.state;
+    return (
+      <div className={styles.container}>
+        <h1 style={{ fontSize: '1.4em', padding: '1rem 2rem' }}>Admin Dashboard</h1>
```
Comment (ahijaouy) [inline frontend/src/components/AdminDash/index.js]: inline CSS should probably be moved to the css module?

File: frontend/src/components/AdminDash/index.js (hunk 4 above)
Comment (Holben888) [reply frontend/src/components/AdminDash/index.js]: Yeah I thought about that syntax but still did it the old way for some reason. I'll change it

File: frontend/src/components/AdminDash/Filters.jsx (hunk 6 above)
Comment (Holben888) [reply frontend/src/components/AdminDash/Filters.jsx]: They do not. this was Kanav's modal using the reactstrap template. Will need to use `formik`

File: frontend/src/components/AdminDash/index.js (hunk 7 above)
Comment (Holben888) [reply frontend/src/components/AdminDash/index.js]: AH forgot to remove this! I had it to experiment with h1 tags in general, and wanted to move that to a global definition for h1.

File: frontend/src/components/AdminDash/AppInfo.jsx (hunk 3 above)
Comment (Holben888) [reply frontend/src/components/AdminDash/AppInfo.jsx]: All information with be in props from the parent container 👍

File: frontend/src/styles/scss/custom.scss (hunk 5 above)
Comment (Holben888) [reply frontend/src/styles/scss/custom.scss]: auto formatting be like that sometimes 🤷‍♀️

**Pull request merged**